shared/health/src/main/java/io/pravega/shared/health/impl/ContributorRegistryImpl.java

```diff
@@ -0,0 +1,309 @@
+            }
+        }
+        // Remove contributor reference from all containers.
+        parents.remove(name);
+        children.remove(name);
+        contributors.remove(name);
+
+        return contributor;
+    }
+
+    /**
+     * Get the {@link HealthContributor} with id 'name'. A NULL value may be returned.
+     *
+     * @param name The name of some {@link HealthContributor} to request from this {@link ContributorRegistry}.
+     * @return The {@link HealthContributor} mapped to by 'name'.
+     */
+    @Override
+    @NonNull
+    synchronized public HealthContributor get(String name) {
+        return contributors.get(name);
+    }
+
+    /**
+     * Provides a reference to the root {@link HealthContributor}.
+     * @return The root {@link HealthContributor}.
+     */
+    @Override
+    synchronized public HealthContributor getRootContributor() {
+        return contributors.get(DEFAULT_CONTRIBUTOR_NAME);
+    }
+
+    /**
+     * Provides the set of dependencies that the {@link HealthContributor} mapped to by 'name' relies on.
+     *
+     * @param name The {@link String} used to query which {@link HealthContributor} to gather it's {@link HealthContributor}
+     *             dependencies.
+     * @return The {@link Collection} used to determine the {@link io.pravega.shared.health.Health}
+     *         of the {@link HealthContributor} mapped to by 'name'.
+     */
+    @Override
+    synchronized public Collection<HealthContributor> dependencies(String name) {
+        return this.children.get(name);
+    }
+
+    /**
+     *
+     * Provides the set of dependencies that the root {@link HealthContributor} relies on.
+     *
+     * @return The {@link Collection} of {@link HealthContributor} used by the root.
+     */
+    @Override
+    synchronized public Collection<HealthContributor> dependencies() {
+        return dependencies(DEFAULT_CONTRIBUTOR_NAME);
+    }
+
+    /**
+     * Supplies a {@link Collection} of *all* {@link HealthContributor} ids (names) registered within
+     * this {@link ContributorRegistry}.
+     *
+     * @return The {@link Collection} of {@link HealthContributor} names.
+     */
+    @Override
+    synchronized public Collection<String> contributors() {
+        return new ArrayList<>(contributors.keySet());
+    }
+
+    /**
+     * Supplies a {@link Collection} of *all* {@link HealthComponent} ids (names) registered within
+     * this {@link ContributorRegistry}. This method returns a subset of the {@link Collection} returned by
+     * {@link ContributorRegistry#contributors()}.
+     *
+     * @return The {@link Collection} of {@link HealthComponent} names.
+     */
+    synchronized public Collection<String> components() {
+        return new ArrayList<>(this.components);
+    }
+
+    /**
+     * Resets the {@link ContributorRegistry} to the state as if it was just instantiated.
+     */
+    synchronized public void clear() {
+        components.clear();
```

Comment:
It's hard for me to check here: do any of these objects that you're cleaning up require actual cleanup (i.e., a `close` method)?

Reply:
No, these are all just standard containers (`Collection`, `Map`, etc).